eng/common/scripts/update-docs-metadata.ps1

```diff
@@ -80,41 +80,53 @@ $pkgs = VerifyPackages -artifactLocation $ArtifactLocation `
   -releaseSha $ReleaseSHA `
   -continueOnError $True
 
-if ($pkgs) {
-  Write-Host "Given the visible artifacts, readmes will be copied for the following packages"
-  Write-Host ($pkgs | % { $_.PackageId })
+$targets = ($Configs | ConvertFrom-Json).targets
 
-  foreach ($packageInfo in $pkgs) {
-    # sync the doc repo
-    $semVer = [AzureEngSemanticVersion]::ParseVersionString($packageInfo.PackageVersion)
-    $rdSuffix = ""
-    if ($semVer.IsPreRelease) {
-      $rdSuffix = "-pre"
-    }
-
-    $readmeName = "$($packageInfo.PackageId.Replace('azure-','').Replace('Azure.', '').Replace('@azure/', '').ToLower())-readme$rdSuffix.md"
-    $readmeLocation = Join-Path $DocRepoLocation $DocRepoContentLocation $readmeName
-
-    if ($packageInfo.ReadmeContent) {
-      $adjustedContent = GetAdjustedReadmeContent -pkgInfo $packageInfo
-    }
-
-    if ($adjustedContent) {
-      try {
-        Push-Location $DocRepoLocation
-        Set-Content -Path $readmeLocation -Value $adjustedContent -Force
+foreach ($config in $targets) {
+  if ($config.mode -eq "Preview") { $includePreview = $true } else { $includePreview = $false }
+  $pkgsFiltered = $pkgs | ? { $_.IsPrerelease -eq $includePreview}
```

Comment:
I am curious whether we always have the `IsPrerelease` config. What if no such kind of information(return empty or null), is it better to have some default case handling?

Reply:
You raise a good point, but I would argue that if a package info can't tell me whether or not it's pre-release reliably, we have major issues elsewhere.